include/oboe/AudioStream.h

```diff
@@ -453,7 +464,8 @@ class AudioStream : public AudioStreamBase {
 private:
     int                  mPreviousScheduler = -1;
 
-    std::atomic<bool>    mDataCallbackEnabled{};
+    std::atomic<bool>    mDataCallbackEnabled{false};
+    std::atomic<bool>    mErrorCallbackCalled{false};
```

Comment:
Do we need to make this atomic? The only time it is mutated is `internalErrorCallback` which is only called by the underlying stream. As such, there should not be any thread safety issues, and it could be replaced with a standard bool without incurring the performance penalty of an atomic operation in a callback method (which blocks close etc.)

Reply:
A lot of Oboe is about trying to protect against bugs in AudioFlinger. It is possible that more than one thread from the stream could call the error callback. The old bug was a race condition that was crashing millions of devices. I want to be very certain there will be no race this time. Plus atomic bool is very fast and this error callback only happens one (or twice). So performance is not an issue.